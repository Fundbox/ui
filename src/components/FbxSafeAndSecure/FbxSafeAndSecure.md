# FbxSafeAndSecure
Safe and Secure logos.
Logos are links that open the relevant service website in a new tab
