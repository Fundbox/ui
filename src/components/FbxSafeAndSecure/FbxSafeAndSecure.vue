<template>
  <article class="safe-and-secure layout-row split vertical-center security-logos">
      <a v-once v-for="img in images" :key="img.ref" class="security-logo" :href="img.ref" target="_blank">
        <img class="security-logo-image" :src="img.src" alt="">
      </a>
    </article>
</template>

<script>
import norton from './assets/security-norton-logo.svg'
import mcafee from './assets/security-mcafee-logo.svg'
import bbb from './assets/security-bbb-logo.png'

export default {
  name: 'FbxSafeAndSecure',
  data() {
    return {
      images: [
        { ref: 'https://trustsealinfo.verisign.com/splash?form_file=fdf/splash.fdf&amp;dn=fundbox.com&amp;lang=en', src: norton },
        { ref: 'http://www.bbb.org/greater-san-francisco/business-reviews/financial-services/fundbox-in-san-francisco-ca-462656#bbbseal', src: bbb },
        { ref: 'https://www.mcafeesecure.com/verify?host=www.fundbox.com', src: mcafee },
      ],
    }
  }
}
</script>

<style lang="scss" scoped>
@import "../../styles/utils/utilities.scss";

.safe-and-secure {
  height: 45px;
}

.security-logo {
  position: relative;
  flex: 0 0 auto;
}

.security-logo-image, .security-logo {
  height: 100%;
}
</style>
