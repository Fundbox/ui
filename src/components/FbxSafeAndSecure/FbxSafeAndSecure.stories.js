import { storiesOf } from '@storybook/vue'
import { withInfo } from 'storybook-addon-vue-info'

import FbxSafeAndSecure from './FbxSafeAndSecure.vue'
import summary from './FbxSafeAndSecure.md'

const stories = storiesOf('Components/SafeAndSecure', module)

stories.add('default', withInfo({ summary })(() => ({
  components: { FbxSafeAndSecure },
  template: `<fbx-safe-and-secure></fbx-safe-and-secure>`
})))
