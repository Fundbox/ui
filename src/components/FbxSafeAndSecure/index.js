import FbxSafeAndSecure from './FbxSafeAndSecure.vue'
import { registerComponents, vueUse } from '../../utils/plugins'

const components = {
  FbxSafeAndSecure
}

const VuePlugin = {
  install(Vue) {
    registerComponents(Vue, components)
  }
}

vueUse(VuePlugin)

export default VuePlugin
export { FbxSafeAndSecure }
