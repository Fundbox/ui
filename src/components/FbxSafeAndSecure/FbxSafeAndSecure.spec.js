import { shallowMount } from '@vue/test-utils'
import FbxSafeAndSecure from './FbxSafeAndSecure.vue'

describe('Components/FbxSafeAndSecure', () => {
  describe('snapshots', () => {
    it('renders the default correctly', () => {
      const wrapper = shallowMount(FbxSafeAndSecure, {
        // mounting options https://vue-test-utils.vuejs.org/api/options.html
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
