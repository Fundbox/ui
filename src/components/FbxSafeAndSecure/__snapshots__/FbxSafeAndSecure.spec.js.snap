// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FbxSafeAndSecure snapshots renders the default correctly 1`] = `<article class="safe-and-secure"><a href="https://trustsealinfo.verisign.com/splash?form_file=fdf/splash.fdf&amp;amp;dn=fundbox.com&amp;amp;lang=en" target="_blank" class="security-logo layout-row vertical-center"><img src="test-file-stub" alt="" class="security-logo-image"></a><a href="http://www.bbb.org/greater-san-francisco/business-reviews/financial-services/fundbox-in-san-francisco-ca-462656#bbbseal" target="_blank" class="security-logo layout-row vertical-center"><img src="[object Object]" alt="" class="security-logo-image"></a><a href="https://www.mcafeesecure.com/verify?host=www.fundbox.com" target="_blank" class="security-logo layout-row vertical-center"><img src="test-file-stub" alt="" class="security-logo-image"></a></article>`;
